### PR TITLE
Add SmartSync Feature to report

### PR DESCRIPTION
**Pull request by ivanbogdanov**

### CODE REVIEW

**Comment by helenren** [comment]
build

**Comment by helenren** [comment]
build

**Comment by helenren** [comment]
build

File: libs/SmartSync/src/com/salesforce/androidsdk/smartsync/app/SmartSyncSDKManager.java [hunk 1]
```diff
@@ -115,6 +118,7 @@ public static void initNative(Context context, KeyInterface keyImpl,
      */
     public static SmartSyncSDKManager getInstance() {
```
Comment (wmathurin) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/app/SmartSyncSDKManager.java]: Any app compiled with smartstore/smartsync will use SmartSyncSDKManager.
Instead you should set that flag from SyncManager.java getInstance

Comment (bhariharan) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/app/SmartSyncSDKManager.java]: +1

**Comment by ivanbogdanov** [comment]
cleaned up SmartSyncSDKManager

**Pull request merged**